### PR TITLE
[SPARK-9596][SQL]treat hadoop classes as shared one in IsolatedClientLoader

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
https://issues.apache.org/jira/browse/SPARK-9596

### CODE REVIEW

**Comment by WangTaoTheTonic** [comment]
The error is: 
>> stderr> FAILED: SemanticException [Error 10072]: Database does not exist: hive_test_db
[info]   stderr> Exception in thread "main" org.apache.spark.sql.execution.QueryExecutionException: FAILED: SemanticException [Error 10072]: Database does not exist: hive_test_db

I should not be produced by this patch.

Jenkins, retest this please.

**Comment by WangTaoTheTonic** [comment]
Jenkins, retest this please.

**Comment by WangTaoTheTonic** [comment]
Jenkins, retest this please.

**Comment by WangTaoTheTonic** [comment]
@marmbrus I am not sure why it will leads to a test failed. Could you help to check this patch and the failed reason? Thanks.

**Comment by marmbrus** [comment]
It looks like your patch is breaking Hive.

**Comment by marmbrus** [comment]
In particular I think that some of hives classes exist in the hadoop namespace so cannot be shared.

**Comment by WangTaoTheTonic** [comment]
@marmbrus Looks like the patch is ok with excluding hive classes. Thanks for your guide :)

**Comment by marmbrus** [comment]
Thanks, merged to master.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala [hunk 1]
```diff
@@ -124,6 +124,7 @@ private[hive] class IsolatedClientLoader(
     name.contains("slf4j") ||
     name.contains("log4j") ||
     name.startsWith("org.apache.spark.") ||
+    (name.startsWith("org.apache.hadoop.") && !name.startsWith("org.apache.hadoop.hive.")) ||
```
Comment (Sephiroth-Lin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala]: scope is too huge, and without reload hadoop, will only have one FileSystem.Cache which will cause [SPARK-11083](https://issues.apache.org/jira/browse/SPARK-11083)

**Pull request closed without merging**